superset/reports/commands/execute.py

```diff
@@ -675,12 +673,13 @@ def __init__(self, task_id: str, model_id: int, scheduled_dttm: datetime):
     def run(self) -> None:
         with session_scope(nullpool=True) as session:
             try:
-                self.validate(session=session)
-                if not self._model:
-                    raise ReportScheduleExecuteUnexpectedError()
-                ReportScheduleStateMachine(
-                    session, self._execution_id, self._model, self._scheduled_dttm
-                ).run()
+                with override_user(_get_user()):
```

Comment:
since this method always runs on a celery async context I see no issue here, but what are we trying to solve?

Reply:
@zhaoyongjie yes it switches the user, just wondering why we need `g.user` set on async tasks, to set `changed_by` fields?

Reply:
The new `BaseFilter` checks `sm.is_admin()`, which in turn checks the roles of `g.user`, which otherwise will be unset. So without this change executing Alerts and Reports fails (the integration tests thankfully picked this up 👍 )